airflow/providers/asana/hooks/asana.py

```diff
@@ -0,0 +1,209 @@
+        elif self.workspace and not (task_params and ("projects" in task_params)):
+            merged_params["workspace"] = self.workspace
+        if task_params:
+            merged_params.update(task_params)
+        return merged_params
+
+    @staticmethod
+    def _validate_create_task_parameters(params: dict) -> None:
+        """Check that user provided minimal create parameters."""
+        required_parameters = {"workspace", "projects", "parent"}
+        if required_parameters.isdisjoint(params):
+            raise ValueError(
+                f"You must specify at least one of {required_parameters} in the create_task parameters"
+            )
+
+    def delete_task(self, task_id: str) -> dict:
+        """Deletes an Asana task."""
+        response = self.client.tasks.delete_task(task_id)  # pylint: disable=no-member
+        return response
+
+    def find_task(self, params: dict) -> list:
+        """
+        Retrieves a list of Asana tasks that match search parameters. For a complete list of parameters,
+        see https://bit.ly/3uIqMj0
+        """
+        merged_params = self._merge_find_task_parameters(params)
+        self._validate_find_task_parameters(merged_params)
+        response = self.client.tasks.find_all(params=merged_params)  # pylint: disable=no-member
+        return list(response)
+
+    def _merge_find_task_parameters(self, search_parameters: dict) -> dict:
+        """Merge find_task parameters with default params from the connection."""
+        merged_params = {}
+        if self.project:
+            merged_params["project"] = self.project
+        # Only use default workspace if user did not provide a project id
+        elif self.workspace and not (search_parameters and ("project" in search_parameters)):
+            merged_params["workspace"] = self.workspace
+        if search_parameters:
+            merged_params.update(search_parameters)
+        return merged_params
+
+    @staticmethod
+    def _validate_find_task_parameters(params: dict) -> None:
+        """Check that user provided minimal search parameters."""
+        one_of_list = {"project", "section", "tag", "user_task_list"}
+        both_of_list = {"assignee", "workspace"}
+        contains_both = both_of_list.issubset(params)
+        contains_one = not one_of_list.isdisjoint(params)
+        if not (contains_both or contains_one):
+            raise ValueError(
+                f"You must specify at least one of {one_of_list} "
+                f"or both of {both_of_list} in the find_task parameters."
+            )
+
+    def update_task(self, task_id: str, params: dict) -> dict:
+        """
+        Updates an existing Asana task. For a complete list of possible parameters, see
+        https://developers.asana.com/docs/update-a-task
+        """
+        response = self.client.tasks.update(task_id, params)  # pylint: disable=no-member
+        return response
+
+    def create_project(self, params: dict) -> dict:
```

Comment:
Please describe the `params` argument in docstring in all methods. The name is vague and having docstring will help users understand what should there be passed